doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -287,123 +279,98 @@ automatically created.
    + :attr:`global` hints are made available by merely requiring the current
      module.
 
-   The various possible :production:`hint_definition`\s are given below.
+   .. todo PR: should the Hint commands be indented or not?
 
-   .. cmdv:: Hint @hint_definition
+   The `Hint` commands are:
 
-      No database name is given: the hint is registered in the ``core`` database.
+   .. cmd:: Hint Resolve {+ {| @qualid | @one_term } } {? @hint_info } {? : {+ @ident } }
+            Hint Resolve {| -> | <- } {+ @qualid } {? @natural } {? : {+ @ident } }
+      :name: Hint Resolve; _
 
-      .. deprecated:: 8.10
+      .. todo PR: not sure what to say if @one_term is provided
 
-   .. cmdv:: Hint Resolve @qualid {? | {? @natural} {? @pattern}} : @ident
-      :name: Hint Resolve
+      .. todo PR: what about @pattern mentioned below??
 
-      This command adds :n:`simple apply @qualid` to the hint list with the head
-      symbol of the type of :n:`@qualid`. The cost of that hint is the number of
-      subgoals generated by :n:`simple apply @qualid` or :n:`@natural` if specified. The
+      .. todo PR: wordy paragraph, maybe should have a little table instead of writing
+         out all the conditions in words?
```

Comment:
I split it into multiple paragraphs, it's a bit more readable now.  The description list format doesn't seem so great given the long clauses "If blah blah blah ".

Is "In this case, the hint is added to the hint list of :tacn:`eauto` instead of the hint list of :tacn:`auto`" referring to different hint databases?  I couldn't guess what was meant.  Also, is the list and the descriptions under "Hint databases defined in the Coq standard library" current?  I only see these in the .v files:

```
Create HintDb core.
Create HintDb rewrite discriminated.
Create HintDb typeclass_instances discriminated.
Create HintDb program discriminated.
Create HintDb ordered_type.
```

Where are the others

Reply:
Regarding hint list of `eauto` vs hint list of `auto`: this is a poor formulation. No, this is the same database, but some hints are ignored by `auto`. The general formulation of the type "the tactic foo bar" is added to the hint database is poor because various automatic tactics (`auto`, `eauto` and `typeclasses eauto`) might use hints in subtly different ways, and it is rarely a direct mapping of user-available tactics. Some hints are not used by `auto` but also some type of hints are only used by `typeclasses eauto`. The only actually separate hint databases are the `Hint Rewrite` databases.

I think the list of databases defined in the standard library is current (maybe not exhaustive). Many databases are defined implicitly / automatically when adding hints to them. `arith`, `bool`, `ord`, `map`, `set`, `fset`, `zarith`, `qarith` seem to be defined that way (found by grepping `"Hint Resolve .* : .*"`).

Reply:
I made it "... the hint is marked for use by :tacn`eauto`"